trio/_core/_run.py

```diff
@@ -1279,12 +1295,14 @@ def _activate_cancel_status(self, cancel_status):
             if self._cancel_status.effectively_cancelled:
                 self._attempt_delivery_of_any_pending_cancel()
 
-    def _attempt_abort(self, raise_cancel):
+    def _attempt_abort(self, raise_cancel: Callable[[], NoReturn]) -> None:
         # Either the abort succeeds, in which case we will reschedule the
         # task, or else it fails, in which case it will worry about
         # rescheduling itself (hopefully eventually calling reraise to raise
         # the given exception, but not necessarily).
-        success = self._abort_func(raise_cancel)
+
+        # TODO: what happens if _abort_func is None?
+        success = self._abort_func(raise_cancel)  # type: ignore
```

Comment:
`_abort_func` can clearly be None - since it's explicitly set to be `None` in a few lines. But this seems like a possibly nasty thing to `# type: ignore`.

Reply:
Searching for `_attempt_abort(`, we only ever call it in the methods immediately below, both of which check that `self._abort_func is not None` first.  I think we'd therefore be safe with a comment along those lines, an `assert self._abort_func is not None, "FATAL INTERNAL ERROR"`, and no type ignore.